### PR TITLE
feat(pie-monorepo): DSW-000 updated components to use Lit 3

### PR DESCRIPTION
**Pull request by siggerzz**
This PR refactors the PIE Web Components & example apps to be compatible with Lit 3.

**Note** - This PR is being merged into a branch called `feature-lit3`. This means that the[ following changesets logic will execute,](https://github.com/justeattakeaway/pie/wiki/Contributing-Guide#feature-versions---next-tag) and when a Version Packages PR is created, it _should_ tag the components as `next`. This means there's no risk of consumers updating to this version until we / they are happy to do so.

---
"pie-monorepo": minor
---

[Changed] - Updated the following dependencies:
- `lit` - `2.8.0` > `3.0.2`
- `typescript` - `5.1.3` > `5.2.2`

[Removed] - Unused `@lit-labs/*` resolutions.
[Fixed] - Bug where `dev` tasks weren't building their depencies where applicable.

---
"wc-next10": minor
"@justeattakeaway/pie-wrapper-react": minor
---

[Changed] - `@lit-labs/react` dependency to `@lit/react` and updated any references.

---
"@justeattakeaway/pie-webc-core"
---

[Changed] - `@validPropertyValues` to support Lit 3.

---
"wc-next10": minor
"wc-nuxt2": minor
---

[Changed] - Webpack config & babel plugins to support Lit 3 as per the upgrade guide - https://lit.dev/docs/releases/upgrade/#using-lit-3-with-webpack-4

---
"@justeattakeaway/pie-icons-react": minor
"pie-storybook": minor
---

[Changed] - `hoistingLimits` config to prevent issues with conflicting dependency versions

---
"wc-nuxt3": minor
---

[Changed] - `nuxt-ssr-lit` dependency from `1.5.1` > `1.6.5`

### CODE REVIEW

File: .changeset/sixty-drinks-fold.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@justeattakeaway/pie-webc-core"
+---
+
+[Changed] - `@validPropertyValues` to support Lit 3.
```
Comment (xander-marjoram) [inline .changeset/sixty-drinks-fold.md:5]:
```suggestion
[Changed] - `@validPropertyValues` and `@requiredProperty` decorators to support Lit 3.
```

File: .changeset/small-buttons-fly.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+"pie-monorepo": minor
+---
+
+[Changed] - Updated the following dependencies:
+- `lit` - `2.8.0` > `3.0.2`
+- `typescript` - `5.1.3` > `5.2.2`
+
+[Removed] - Unused `@lit-labs/*` resolutions.
+[Fixed] - Bug where `dev` tasks weren't building their depencies where applicable.
```
Comment (xander-marjoram) [inline .changeset/small-buttons-fly.md:10]:
```suggestion
[Fixed] - Bug where `dev` tasks weren't building their dependencies where applicable.
```

File: packages/tools/pie-icons-webc/package.json [hunk 3]
```diff
@@ -64,7 +64,7 @@
     "@web/test-runner": "0.16.1",
     "just-kebab-case": "4.2.0",
     "just-pascal-case": "3.2.0",
-    "lit": "2.7.5",
+    "lit": "3.0.2",
```
Comment (xander-marjoram) [inline packages/tools/pie-icons-webc/package.json:67]: This is fine for now but a good candidate for bundling improvements

**Review by xander-marjoram — APPROVED** [review]

**Review by ashleynolan — APPROVED** [review]

File: apps/examples/wc-next10/package.json [hunk 4]
```diff
@@ -31,5 +31,11 @@
   },
   "installConfig": {
     "hoistingLimits": "workspaces"
+  },
+  "devDependencies": {
+    "@babel/plugin-transform-logical-assignment-operators": "7.22.11",
+    "@babel/plugin-transform-nullish-coalescing-operator": "7.22.11",
+    "@babel/plugin-transform-optional-chaining": "7.23.0",
+    "babel-loader": "8"
```
Comment (raoufswe) [inline apps/examples/wc-next10/package.json:39]: @siggerzz was that meant to be 8.0.0 or it's valid to specify a version like this 👀

**Pull request merged**